FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/Dicom/DicomApiDataClient.cs

```diff
@@ -0,0 +1,190 @@
+        {
+            try
+            {
+                var searchRequest = new HttpRequestMessage(HttpMethod.Get, uri);
+                if (accessToken != null)
+                {
+                    // Currently we support accessing service endpoints with Managed Identity.
+
+                    // The thread-safe AzureServiceTokenProvider class caches the token in memory and retrieves it from Azure AD just before expiration.
+                    // https://docs.microsoft.com/en-us/dotnet/api/overview/azure/service-to-service-authentication#using-the-library
+                    searchRequest.Headers.Authorization = new AuthenticationHeaderValue("Bearer", accessToken);
+                }
+
+                HttpResponseMessage response = await _httpClient.SendAsync(searchRequest, cancellationToken);
+                response.EnsureSuccessStatusCode();
+
+                _logger.LogInformation("Successfully retrieved result for url: '{url}'.", uri);
+
+                return await response.Content.ReadAsStringAsync(cancellationToken);
+            }
+            catch (HttpRequestException hrEx)
+            {
+                switch (hrEx.StatusCode)
+                {
+                    case HttpStatusCode.Unauthorized:
+                        _diagnosticLogger.LogError(string.Format("Failed to search from server: Server {0} is unauthorized.", _dicomApiDataSource.ServerUrl));
+                        _logger.LogInformation(hrEx, "Failed to search from server: Server {0} is unauthorized.", _dicomApiDataSource.ServerUrl);
+                        break;
+                    case HttpStatusCode.NotFound:
+                        _diagnosticLogger.LogError(string.Format("Failed to search from server: Server {0} is not found.", _dicomApiDataSource.ServerUrl));
+                        _logger.LogInformation(hrEx, "Failed to search from server: Server {0} is not found.", _dicomApiDataSource.ServerUrl);
+                        break;
+                    case HttpStatusCode.Forbidden:
+                        _diagnosticLogger.LogError(string.Format("Failed to search from server: Server {0} is forbidden.", _dicomApiDataSource.ServerUrl));
+                        _logger.LogInformation(hrEx, "Failed to search from server: Server {0} is forbidden.", _dicomApiDataSource.ServerUrl);
+                        break;
+                    default:
+                        _diagnosticLogger.LogError(string.Format("Failed to search from server: Status code: {0}. Reason: {1}", hrEx.StatusCode, hrEx.Message));
+                        _logger.LogInformation(hrEx, "Failed to search from server: Status code: {0}. Reason: {1}", hrEx.StatusCode, hrEx.Message);
+                        break;
+                }
+
+                throw new ApiSearchException(
+                    string.Format(Resource.FhirSearchFailed, uri, hrEx.Message),
```

Comment:
Create a new string Resource, such as `DicomSearchFailed`, instead of using `FhirSearchFailed`

Reply:
Fixed, thanks!